### DISCUSSION TITLE
sf_set_chunk() question

### DISCUSSION THREAD

**Discussion by TrixieDev** [libsndfile/libsndfile#911] [Q&A]
Googling (and asking Erik) didn't provide any answers, so I'll try my luck and ask here. Is writing custom chunks through sf_set_chunk() only possible for RIFF files, or would the function also work on AIFF or IFF-8SVX?

**Comment by evpobr** [comment]
Hi @TrixieDev . It should work for AIFF, CAF, RF64 and WAV.

**Reply by TrixieDev** [reply]
I see. OK, thanks! Too bad: I was hoping for 8SVX as well :-) My problem is that libsndfile doesn't support writing stereo 8SVX files because the format expects stereo data to be stored as non-interleaved (one channel following the other). I was told libsndfile doesn't support this for design reasons, which I understand. But if sf_set_chunk() could be used on 8SVX, I'd be able to open the file for writing, then use sf_set_chunk() to add the CHAN chunk (overriding the default mono), and then use the library to separately write the left channel data followed by the right channel data. Would be way easier than implementing my own, non-sndfile stereo 8SVX write code.
